salt/modules/virt.py

```diff
@@ -1364,21 +1364,22 @@ def init(
 
                      .. versionadded:: 2019.2.0
     :param boot:
-        Specifies kernel for the virtual machine, as well as boot parameters
-        for the virtual machine. This is an optionl parameter, and all of the
-        keys are optional within the dictionary. If a remote path is provided
-        to kernel or initrd, salt will handle the downloading of the specified
-        remote fild, and will modify the XML accordingly.
+        Specifies kernel, initial ramdisk and kernel command line parameters for the virtual machine.
+        This is an optional parameter, all of the keys are optional within the dictionary. If a
+        remote path is provided to kernel or initrd, salt will handle the downloading of the specified
+        remote file and modify the XML accordingly. To boot VM with UEFI, specify loader and nvram path.
 
         .. code-block:: python
 
             {
                 'kernel': '/root/f8-i386-vmlinuz',
                 'initrd': '/root/f8-i386-initrd',
-                'cmdline': 'console=ttyS0 ks=http://example.com/f8-i386/os/'
+                'cmdline': 'console=ttyS0 ks=http://example.com/f8-i386/os/',
+                'loader': '/usr/share/OVMF/OVMF_CODE.fd',
+                'nvram': '/usr/share/OVMF/OVMF_VARS.ms.fd'
             }
 
-        .. versionadded:: 3000
+        .. versionadded:: sodium
```

Comment:
There are two points that need more rework:
  * this blob of doc needs to be maintained in 4 places... which will be a pain later
  * the `kernel`, `initrd` and `cmdline` parameters have been added in `3000` we need to keep that info around.

Here is how I view things:

```suggestion
    :param boot:
        Specifies kernel for the virtual machine, as well as boot parameters
        for the virtual machine. This is an optional parameter, and all of the
        keys are optional within the dictionary. The structure of the dictionary
        is documented in :ref:`init-boot-def`
        
        .. versionadded: 3000
        
        .. _init-boot-def:
        
        .. rubric:: Boot parameters definition
        
        The boot parameters dictionary can contains the following properties:
        
        kernel:
            The URL or path to the kernel to run the virtual machine with.
        
        initrd:
            The URL or path to the initrd file to run the virtual machine with.
        
        cmdline:
            The parameters to pass to the kernel provided in the `kernel` property.
            
        loader:
            The path to the UEFI binary loader to use.
            
            .. versionadded: sodium
            
        nvram:
            The path to the UEFI data template. The file will be copied when creating the virtual machine.
            
            .. versionadded: sodium
            
        If a remote path is provided to ``kernel`` or ``initrd``, it will be downloaded from the specified
        remote file, and will modify the XML accordingly. 
        
        .. code-block:: python
            {
                'kernel': '/root/new-vmlinuz',
                'initrd': '/root/new-initrd',
                'cmdline': 'console=ttyS0 ks=http://example.com/f8-i386/os/',
                'loader': '/usr/share/new/OVMF_CODE.fd',
                'nvram': '/usr/share/new/OVMF_VARS.ms.fd'
            }
```